lib/sanity/attributable.rb

```diff
@@ -22,33 +29,46 @@ def attributes
         @attributes ||= []
       end
 
-      def default_attributes
-        @defaults ||= {}
+      def aliased_attributes
+        @aliased_attributes ||= {}
       end
 
       private
 
-      def attribute(name, default: nil)
-        attributes << name
-        default_attributes.merge!("#{name}": default)
+      def attribute(name, default: nil, as: nil, type: nil)
+        attributes.push(name)
+
+        self.define_method("#{name}=") do |value|
+          @attributes[name] =
+            case type
+            when :boolean
+              value.to_s == 'true'
+            when :float
+              value.to_f
+            when :integer
+              value.to_i
+            when :time
+              Time.parse(value.to_s)
+            else
+              value
+            end
+        end
+        self.define_method("#{name}") { @attributes[name] }
+
+        if as
+          aliased_attributes[as] = name
+          self.alias_method("#{as}=", "#{name}=")
+          self.alias_method(as, name)
+        end
       end
+
     end
 
     attr_reader :attributes
```

Comment:
This is broken now as well since we're no longer setting the instance variable. This will always return nil

Reply:
It's being set on line 42 in the setter. This still works.

```ruby
[8] pry(main)> person = Person.new(_id: 123)
=> #<Person:0x1640 _id: 123>
[9] pry(main)> person.attributes
=> {:_id=>123}
[10] pry(main)>
```